warl_proposal/csr_template_draft1.yaml

```diff
@@ -0,0 +1,130 @@
+############################################ Template description ############################################
+# csr template for csrs with subfields
+<name>:                                   # name of the csr
+  description: <text>                     # textual description of the csr
+  address: <hex>                          # address of the CSR
+  priv_mode: <D/M/H/S/U>                  # privilege mode that owns the register
+  reset-val: <hex>                        # Reset value of the register. This an accumulation of the all reset values of the 
+                                          # sub-field 
+  rv32:                                   # this node and its subsequent fields can exist if [M/S/U]XL value can be 1
+    exists: <boolean>                     # indicates if the csr exists in rv32 mode or not. When False, all fields below 
+                                          # will be trimmed off in the checked yaml. False also indicates that 
+                                          # access-exception should be generated
+    subfields:                            # a quick summary of the list of all subfields of the csr including a list of 
+                                          # WPRI fields of the csr.
+      - <subfield_name1>
+      - <subfield_name2>
+      - wpri                              # contains a list of all WPRI fields within the csr. Can be empty as well
+        - [30-23]
+        - 6
+    <subfield_name1>:                     # name of the subfield
+      description: <text>                 # textual description of the csr
+      msb: <hex>                          # msb index of the subfield. max: 31, min:0
+      lsb: <hex>                          # lsb index of the subfield. min: 31, min:0
+      implemented: <boolean>              # indicates if the user has implemented this subfield or not. When False, all 
+                                          # fields below this will be trimmed.
+      reset-val: <hex>                    # reset value of the subfield.
+      type:                               # indicates the type of the field. Can be one of the following only:
+        RO: <hex>                         # indicates the field is readonly with this value. This value should be the same 
+                                          # as the reset-value.
+        WARL:                             # indicates this field is WARL type. Refer to warl_proposal for more info.
+          dependency_variables: [list]    
+          behavior:
+            - func1:
+                dependency_values: [list] 
+                warl-func:
+                  mode: <string>          
+                  legal: [list]           
+        WLRL: [list of ranges]            # indicates this field is WLRL and the list of legal values as defined by the spec
+  rv64:                                   # this node and its subsequent fields can exist if [M/S/U]XL value can be 2
```

Comment:
for rv32/rv64/rv128, we are leaning to remove it from the template. Instead, we could have individual template for RV32, RV64 and RV128. User can base on the XLEN to choose on to implement. This could make the format simpler, what do you think?

Reply:
This is primarily to support multiple values that MXL can take in MISA. 
For example consider the case that MXL can take values 1 and 2. Now for mstatus SXL and UXL field (which are WARL) do not exist in rv32 mode but they do exist in rv64 mode. So the same csr has different behavior in rv32/rv64 modes. We felt it would be better to describe it this way.

Another example would be the satp register which has different subfield sizes in rv32 and rv64 mode. In each mode the WARL function itself can also be different.

Reply:
Sounds good, as long as user only need to input one based on XLEN

Reply:
The user will have to define the behavior in both rv32 and rv64 modes if MXL can take values 1 and 2. If an implementation hardwires MXL to just 1 or just 2 then only rv32 or rv64 modes need to be defined respectively.

Reply:
Sounds good.